tests/variable/variable.cpp

```diff
@@ -1699,3 +1699,52 @@ TEST(Variable, variableInterfaceDontDowngrade)
     EXPECT_EQ("", v3->interfaceType());
     EXPECT_EQ("public", v4->interfaceType());
 }
+
+TEST(Variable, addVariableDuplicates)
+{
+    auto model = libcellml::Model::create("model");
+    auto tomato = libcellml::Component::create("tomato");
+    auto apple = libcellml::Component::create("apple");
+    auto pip = libcellml::Variable::create("pip");
+
+    EXPECT_TRUE(model->addComponent(tomato));
+    EXPECT_TRUE(model->addComponent(apple));
+
+    // Adding a pip to the tomato
+    EXPECT_TRUE(tomato->addVariable(pip));
+
+    // Add some more ... (this shouldn't be allowed!)
+    EXPECT_FALSE(tomato->addVariable(pip));
+
+    // Add a pip to the apple as well (I'd expect this to move the variable out
+    // of the tomato component, but it just adds it here too):
+    EXPECT_TRUE(apple->addVariable(pip));
+
+    // I would expect the last addVariable (which was in to the apple component) to move the
+    // pip variable there.
+    EXPECT_EQ(size_t(1), apple->variableCount());
+    // ... and out of the tomato:
+    EXPECT_EQ(size_t(0), tomato->variableCount());
+}
+
+TEST(Variable, addEquivalenceReturnsFalseFalse)
```

Comment:
`addEquivalenceReturnsFalseFalse`? Why `False` twice?

Reply:
Because it falsely returns false ;) It had (before the fix) changed the model but said it didn't.

Reply:
Not the case anymore though...?

Reply:
Done in 2b2884ad